### PR TITLE
[COM-144][COM-147][COM-31][COM-158 - COM-154][COM-152 - COM-149][COM-27] Entire system validation, camera orbit, sciprting system, tons of fixes, integration.

### PR DESCRIPTION
**Pull request by le-rich**

### CODE REVIEW

**Review by gravity-projects — APPROVED** [review]
:3

**Review by jharris99-git — APPROVED** [review]
:3

File: Rendering/src/Renderer.cpp [hunk 1]
```diff
@@ -48,15 +90,47 @@ void Renderer::Update()
...
+
+        if (meshComponent != nullptr)
+        {
+            glm::vec3 localScale = entity->transform->getLocalScale();
+
+            glm::mat4 modelMatrix = glm::translate(glm::mat4(1.0f), entity->transform->getWorldPosition()) *
+                                    glm::mat4_cast(entity->transform->getLocalOrientation()) *
+                                    glm::scale(glm::mat4(1.0f), localScale);
+
+            mEngineUniformBuffer.updateMatrices(
```
Comment (Tissure) [inline Rendering/src/Renderer.cpp:111]: Redundant update. 
We can set the camera UBO before we enter the loop. 
After we can change the model via SetSubData() like on line 62.

**Review by Tissure — APPROVED** [review]
Minor comment. lg otherwise

**Pull request merged**